lib/workers/bin/run_single_worker.rb

```diff
@@ -109,8 +109,10 @@ def all_role_names
 
   begin
     runner_options[:guid] = worker.guid
-    worker.class::Runner.new(runner_options).tap(&:setup_sigterm_trap).start
```

Comment:
I could be wrong, but this might have been set up this way to allow testing of the .start method in rspec, without it setting up the signal traps, which would otherwise interfere with RSpec's own signal traps.  I'd expect that if we move `setup_sigterm_trap` into start, then there should be some stubbing of that method somewhere.

Reply:
I see below you've stubbed the Process.kill calls, so that implies to me that our signal are enabled, thus cloberring RSpec's signal handling.

Reply:
I can just revert this bit back to the `.tap` that we have here.

Every worker starting method goes through this script now so there's not much of a difference.

Reply:
I'm not convinced this is related to the stubbing I had to to below though.

We don't actually run the `Runner#start` method in those specs so I really don't think we set up any of the traps either way. I will say I was surprised that I had to stub out the `kill -9` though because that should have been being called before my changes (I thought it would have already been stubbed).